apps/prairielearn/src/ee/lib/ai-grading.sql

```diff
@@ -10,6 +10,17 @@ WHERE
   AND iq.requires_manual_grading
   AND iq.status != 'unanswered';
 
+-- BLOCK select_all_instance_questions
+SELECT
+  iq.*
+FROM
+  instance_questions AS iq
+  JOIN assessment_instances AS ai ON (ai.id = iq.assessment_instance_id)
+WHERE
+  ai.assessment_id = $assessment_id
```

Comment:
Is filtering on this necessary? This won't narrow things any further than the check for `assessment_question_id` below.

Consider naming this query `select_instance_questions_for_assessment_question` instead to make it clear that it is not _all_ instance questions.

Reply:
I copied the skeleton for this function from the assessmentQuestion.sql file where it had both filtering on assessment question id and assessment id. I thought this was necessary to avoid some weird situations that I couldn't think of. If this is not needed, I would happily remove this filtering.

Reply:
Update: I've removed the filter on assessment id (and thus removed the join on `assessment_instances`). In the PR for the mvp, I was using a the same set of filtering on both `assessment_id` and `assessment_question_id` in `select_instance_questions_manual_grading`. Should I remove the filter on `assessment_id` as well for that in this PR?

Reply:
That'd probably be best for consistency. The only time it should be necessary to query based on both `assessment_id` and `assessment_question_id` is if `assessment_question_id` is untrusted, in which case you'd want to verify membership to a particular assessment (assuming that `assessment_id` is trusted). From what I can tell, we're always dealing with a trusted `assessment_question_id` here.

Reply:
Just updated the other query as well